sdk/eventhub/azure-eventhubs/azure/eventhub/aio/client_async.py

```diff
@@ -85,33 +97,31 @@ def _create_auth(self, username=None, password=None):
                                                 get_jwt_token, http_proxy=http_proxy,
                                                 transport_type=transport_type)
 
+    async def _handle_exception(self, exception, retry_count, max_retries):
+        await _handle_exception(exception, retry_count, max_retries, self, log)
+
+    async def _close_connection(self):
+        self._conn_manager.reset_connection_if_broken()
+
     async def _management_request(self, mgmt_msg, op_type):
-        alt_creds = {
-            "username": self._auth_config.get("iot_username"),
-            "password": self._auth_config.get("iot_password")}
-        connect_count = 0
+        max_retries = self.config.max_retries
+        retry_count = 0
         while True:
-            connect_count += 1
-            mgmt_auth = self._create_auth(**alt_creds)
+            mgmt_auth = self._create_auth()
             mgmt_client = AMQPClientAsync(self.mgmt_target, auth=mgmt_auth, debug=self.config.network_tracing)
             try:
-                await mgmt_client.open_async()
+                conn = await self._conn_manager.get_connection(self.host, mgmt_auth)
+                await mgmt_client.open_async(connection=conn)
                 response = await mgmt_client.mgmt_request_async(
                     mgmt_msg,
                     constants.READ_OPERATION,
                     op_type=op_type,
                     status_code_field=b'status-code',
                     description_fields=b'status-description')
                 return response
-            except (errors.AMQPConnectionError, errors.TokenAuthFailure, compat.TimeoutException) as failure:
-                if connect_count >= self.config.max_retries:
-                    err = ConnectError(
-                        "Can not connect to EventHubs or get management info from the service. "
-                        "Please make sure the connection string or token is correct and retry. "
-                        "Besides, this method doesn't work if you use an IoT connection string.",
-                        failure
-                    )
-                    raise err
+            except Exception as exception:
+                await self._handle_exception(exception, retry_count, max_retries)
+                retry_count += 1
             finally:
                 await mgmt_client.close_async()
```

Comment:
This is an existing bug in the track1 SDK - it should to be:
```
if mgmt_client:
    await mgmt_client.close_async()
```

Reply:
I put mgmt_client definition out of try. It will only throw an exception unless there is a bug